resources/repo.rb

```diff
@@ -1,28 +1,19 @@
-property :repo_name, String, desired_state: false, identity: true, name_attribute: true
-property :repo_type, String, desired_state: false, identity: true, default: 'maven2-hosted'
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven2-hosted'
 property :attributes, Hash, default: lazy { ::Mash.new } # Not mandatory but strongly recommended in the generic case.
 property :online, [true, false], default: true
-property :api_endpoint, String, desired_state: false, identity: true,
-                                default: lazy { node['nexus3']['api']['endpoint'] }
-property :api_username, String, desired_state: false, identity: true,
-                                default: lazy { node['nexus3']['api']['username'] }
-property :api_password, String, desired_state: false, identity: true, sensitive: true,
+property :api_endpoint, String, desired_state: false, default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, desired_state: false, default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, desired_state: false, sensitive: true,
                                 default: lazy { node['nexus3']['api']['password'] }
 
 load_current_value do |desired|
-  api_endpoint desired.api_endpoint
-  api_username desired.api_username
-  api_password desired.api_password
-
-  def apiclient
-    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
-  end
-
   begin
-    res = apiclient.run_script('get_repo', desired.repo_name)
+    res = ::Nexus3::Api.new(api_endpoint, api_username, api_password).run_script('get_repo', desired.repo_name)
     current_value_does_not_exist! if res == 'null'
```

Comment:
Handle this case with `config.nil?` after `JSON.parse`

Reply:
done